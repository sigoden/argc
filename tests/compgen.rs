use super::SPEC_SCRIPT;

const HELP_TAG_SCRIPT: &str = r#"
# @help Print help information

# @cmd
foo() { :; }

# @cmd
bar() { :; }
"#;

#[test]
fn test_compgen() {
    snapshot_compgen!(SPEC_SCRIPT, "");
}

#[test]
fn test_compgen_help() {
    snapshot_compgen!(SPEC_SCRIPT, "help");
}

#[test]
fn test_compgen_subcommand() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names ");
}

#[test]
fn test_compgen_option_choices() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 ");
}

#[test]
fn test_compgen_option_choices2() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 a");
}

#[test]
fn test_compgen_option_choices3() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 a ");
}

#[test]
fn test_compgen_positional() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires ");
}

#[test]
fn test_compgen_positional_arg() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires arg1 ");
}

#[test]
fn test_compgen_positional_arg2() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires arg1 arg2 ");
}

#[test]
fn test_compgen_positional_choices() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_with_choices ");
}

#[test]
fn test_compgen_help_tag() {
    snapshot_compgen!(HELP_TAG_SCRIPT, "");
}

#[test]
fn test_compgen_help_tag2() {
    snapshot_compgen!(HELP_TAG_SCRIPT, "help");
}

#[test]
fn test_compgen_choice_fn() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --op11 ");
}
