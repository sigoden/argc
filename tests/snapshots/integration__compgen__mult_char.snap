---
source: tests/compgen.rs
expression: data
---
************ COMPGEN Bash `prog --oa abc,` ************
abc
def
ijk

************ COMPGEN Elvish `prog --oa abc,` ************
abc,abc	0	abc	
abc,def	0	def	
abc,ijk	0	ijk	

************ COMPGEN Fish `prog --oa abc,` ************
abc,abc
abc,def
abc,ijk

************ COMPGEN Nushell `prog --oa abc,` ************
abc,abc
abc,def
abc,ijk

************ COMPGEN Powershell `prog --oa abc,` ************
'abc,abc'	0	abc	
'abc,def'	0	def	
'abc,ijk'	0	ijk	

************ COMPGEN Xonsh `prog --oa abc,` ************
abc,abc	0	abc	
abc,def	0	def	
abc,ijk	0	ijk	

************ COMPGEN Zsh `prog --oa abc,` ************
abc,abc	abc
abc,def	def
abc,ijk	ijk


